### PR TITLE
tests/specs: elliminating #getKLabel

### LINKED ISSUE #725: HKG with Haskell - Lemmas that use reflection
The Haskell backend does not handle reflection and crashes.

https://github.com/kframework/evm-semantics/blob/master/tests/specs/erc20/abstract-semantics-segmented-gas.k#L20
https://github.com/kframework/evm-semantics/blob/master/tests/specs/erc20/abstract-semantics-segmented-gas.k#L35
https://github.com/kframework/evm-semantics/blob/master/tests/specs/erc20/verification.k#L14

### PR DESCRIPTION
**Pull request by denis-bogdanas**
Fixes #725

### CODE REVIEW

File: tests/specs/erc20/verification.k [hunk 1]
```diff
@@ -12,7 +12,10 @@ module VERIFICATION
 
     //Rules for #padToWidth with non-regular symbolic arguments.
     rule #padToWidth(32, #asByteStack(V)) => #buf(32, V)
-      requires 0 <=Int V andBool V <Int pow256 andBool #notKLabel(V, "#asWord")
+      requires 0 <=Int V andBool V <Int pow256
+
+    rule #buf(N, #asWord(WS)) => WS
+      requires #noOverflow(WS) andBool N ==Int #sizeByteArray(WS)
```
Comment (ehildenb) [inline tests/specs/erc20/verification.k]: This lemma can go in `tests/specs/lemmas.k` instead I think. Make sure to mark it with `[simplification]` for the Haskell backend.

File: tests/specs/bihu/verification.k [hunk 2]
```diff
@@ -12,7 +12,10 @@ module VERIFICATION
 
     //Rules for #padToWidth with non-regular symbolic arguments.
     rule #padToWidth(32, #asByteStack(V)) => #buf(32, V)
-      requires 0 <=Int V andBool V <Int pow256 andBool #notKLabel(V, "#asWord")
+      requires 0 <=Int V andBool V <Int pow256
+
+    rule #buf(N, #asWord(WS)) => WS
+      requires #noOverflow(WS) andBool N ==Int #sizeByteArray(WS)
```
Comment (ehildenb) [inline tests/specs/bihu/verification.k]: This lemma duplicated.

File: tests/specs/benchmarks/verification.k [hunk 3]
```diff
@@ -182,7 +182,10 @@ module VERIFICATION
     rule #asWord(#buf(32, DATA)) => DATA  requires #rangeUInt(256, DATA)
 
     rule #padToWidth(32, #asByteStack(V)) => #buf(32, V)
-      requires 0 <=Int V andBool V <Int pow256 andBool #notKLabel(V, "#asWord")
+      requires 0 <=Int V andBool V <Int pow256
+
+    rule #buf(N, #asWord(WS)) => WS
+      requires #noOverflow(WS) andBool N ==Int #sizeWordStack(WS)
```
Comment (ehildenb) [inline tests/specs/benchmarks/verification.k]: This lemma duplicated.

**Comment by yzhang90** [comment]
LGTM

**Comment by ehildenb** [comment]
Please fix the lemma duplication, then we can merge.

**Pull request merged**